### PR TITLE
ADRG: modified to correctly identify the format of SRP .gen files (fixes #953)

### LINKED ISSUE #953: Multi image ASRP datasets open using the GEN file are incorrectly identified as ADRG
## Problem Description.

When the OPEN_GEN option is enabled and an ASRP dataset containing multiple images is opened using the GEN file the dataset is incorrectly identified as ADRG.

## Steps to reproduce the problem.

```
C:\>gdalinfo TPCE1B01.GEN
Driver: ADRG/ARC Digitized Raster Graphics
Files: TPCE1B01.GEN
Size is 512, 512
Coordinate System is:
GEOGCS["WGS 84",
    DATUM["WGS_1984",
        SPHEROID["WGS 84",6378137,298.257223563]],
    PRIMEM["Greenwich",0],
    UNIT["degree",0.0174532925199433],
    AUTHORITY["EPSG","4326"]]
Subdatasets:
  SUBDATASET_1_NAME=ADRG:TPCE1B01.GEN,.\TPCE1B01.IMG
  SUBDATASET_1_DESC=ADRG:TPCE1B01.GEN,.\TPCE1B01.IMG
  SUBDATASET_2_NAME=ADRG:TPCE1B01.GEN,.\TPCE1B02.IMG
  SUBDATASET_2_DESC=ADRG:TPCE1B01.GEN,.\TPCE1B02.IMG
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0,  512.0)
Upper Right (  512.0,    0.0)
Lower Right (  512.0,  512.0)
Center      (  256.0,  256.0)

```

Simple ASRP dataset containing 2 arc zone images:
[TPCE1B01.zip](https://github.com/OSGeo/gdal/files/2427806/TPCE1B01.zip)

## Suggested Solution

Modify the ADRG driver to differentiate between ADRG and ASRP datasets by checking for the existence of a NWO subfield in the GEN field and returning a null pointer if it is not found (SRP datasets do not contain the NWO subfield). This can be done by adding the following code to the function ADRGDataset::GetIMGListFromGEN, immediately after the statement "if ( strcmp(RTY, "GIN") != 0 ) continue;":

```
/* test for existence of a NWO subfield in the GEN field */
            const char* NWO = record->GetStringSubfield("GEN", 0, "NWO", 0);
            if( NWO == nullptr )
                return nullptr;

```
## Operating system

Windows 7 and 10

## GDAL version and provenance

GDAL 1.11.1 and later

### PR DESCRIPTION
**Pull request by sab56**
## What does this PR do?
Adds checks to ensure .gen files are correctly identified as ADRG or SRP.

## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

Provide environment details, if relevant:

* OS:
* Compiler:

### CODE REVIEW

File: gdal/frmts/adrg/srpdataset.cpp [hunk 1]
```diff
@@ -36,8 +36,8 @@
 #include <cstdlib>
 #include <algorithm>
 
-// Uncomment to recognize also .gen files in addition to .img files
-// #define OPEN_GEN
+// Enable option to recognize .gen files in addition to .img files
```
Comment (rouault) [inline gdal/frmts/adrg/srpdataset.cpp]: I'd prefer if we keep it disabled. This is a bit annoying that the same dataset is recognized by two separate files (thinking of the context of a GUI explorer like QGIS)

Comment (sab56) [reply gdal/frmts/adrg/srpdataset.cpp]: I don't understand why transmittal header files (.thf) are enabled by default for both formats but .gen files aren't. There shouldn't be a problem as long as the drivers can differentiate between a ADRG and SRP .gen file.

Comment (rouault) [reply gdal/frmts/adrg/srpdataset.cpp]: My point was that if OPEN_GEN is enabled then when exploring a directory with foo.gen and foo.img from a ASRP/USRP datasets, both files will be recognized, which mean 2 entries in QGIS browser tree for example. Enabling OPEN_GEN should belong to a different change (I don't recall all the rationale for this hesitation between .gen and .img. It has been a long time...)
THF files are a bit different as there might point to several subdatasets.

**Comment by rouault** [comment]
Changes in adrgdataset.cpp only pushed per 8ef93c94be1c3ed8d039dff1c9eec112551aa6e1

**Pull request closed without merging**